pkg/scheduler/scheduler_test.go

```diff
@@ -3043,8 +3043,238 @@ func TestSchedule(t *testing.T) {
 				"eng-alpha/a1-admitted": *utiltesting.MakeAdmission("ClusterQueueA").Assignment("gpu", "on-demand", "1").Obj(),
 			},
 		},
-	}
 
+		"wl1 with on-demand flavor succeeds": {
```

Comment:
@Horiodino I tested that all these cases still pass if `attemptedFlavorIdx` is moved back (https://github.com/kubernetes-sigs/kueue/pull/2407 reverted) to https://github.com/kubernetes-sigs/kueue/pull/2407/files#diff-0870ec2c0ea330a7d087a6ef599905bf9b7164a0dd889d1b00ba17c31a1bfd05L430

I think we need a test which would actually show that moving https://github.com/kubernetes-sigs/kueue/pull/2407/files#diff-0870ec2c0ea330a7d087a6ef599905bf9b7164a0dd889d1b00ba17c31a1bfd05R407 solved some issue.

Reply:
Hi @Horiodino, sorry to ping you. Will you have time to address this recently? I am asking because my PR #4662 is on hold for this one.